src/components/LocalLink/index.js

```diff
@@ -1,28 +1,20 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 
-import NextLink from 'next/link'
-
-import { PAGE_DOC } from '../../consts'
+import GatsbyLink from 'gatsby-link'
 
 export default function LocalLink({
   children,
-  as: Component,
-  href,
+  as: SC,
+  href: to,
   ...restProps
 }) {
-  const nextProps = href.match(/^\/doc/)
-    ? { href: PAGE_DOC, as: href }
-    : { href }
+  const Component = SC ? SC.withComponent(GatsbyLink) : GatsbyLink
```

Comment:
I'm sorry to say this API is [being deprecated](https://styled-components.com/docs/api#withcomponent)!

The new way to do this is to pass an `as` attribute to the StyledComponent itself as described [here](https://styled-components.com/docs/api#as-polymorphic-prop).

Reply:
Good catch! I missed this part on the docs somehow. Will change it.